--- a/Model/Currency.php
+++ b/Model/Currency.php
@@ -365,10 +365,6 @@ class Currency extends \Magento\Framework\Model\AbstractModel
          */
         $price = sprintf("%F", $price);

-        if ($this->canUseNumberFormatter($options)) {
-            return $this->formatCurrency($price, $options);
-        }
-
         return $this->_localeCurrency->getCurrency($this->getCode())->toCurrency($price, $options);
     }
